x/merkledb/codec.go

```diff
@@ -75,22 +74,12 @@ type decoder interface {
 }
 
 func newCodec() encoderDecoder {
```

Comment:
I think we're going to want to completely remove this type (now that it's an empty unexported struct). But I think we can leave that to a followup.